CHANGELOG.md

```diff
@@ -3,6 +3,117 @@ All notable changes to this project will be documented in this file.
+  - Replaced some of alpaka's metafunctions with their standard counterparts #1501
+  - Make use of C++17 mandatory copy elision #1502
+  - Simplified CPU kernel launches #1511
+  - Make use of generic `std` container interfaces #1554
+  - Replaced `std::enable_if` with `if constexpr` where possible #1556
+  - Replaced `alpaka::ignore_unused` with C++17 `[[maybe_unused]]` and `std::ignore` #1563
+  - Make use of nested namespaces #1587 #1592
+  - Make use of variable template versions of `std` traits #1594
+- alpaka `Event`s can now be queried for their device type #1479
+- Some alpaka buffers can now be allocated asynchronously within a device queue (queue-ordered memory buffers) #1481
+  - This capability can be queried with the `hasAsyncBufSupport` trait #1578
+- alpaka buffers can now be zero-dimensional (scalar) #1536
+- Apply `alpaka::memset` and `alpaka::memcpy` to the whole buffer if no extent is supplied by the user #1547
+- Added an accessor-like interface to buffers and views #1570
+- Host code utilizing the CUDA and HIP back-ends can now be compiled with a non-CUDA/HIP compiler if there is no device code in the translation unit #1567
+- Added `alpaka::getNativeHandle()` to obtain the back-end specific handles from alpaka `Device`s #1579
+  - `alpaka::getNativeHandle()` can also be called on `Queue`s and `Event`s #1623
+- Added an experimental SYCL back-end. All SYCL back-end functionality currently lives in the `alpaka::experimental` namespace. See the `README_SYCL.md` for more information about the usage and the restrictions of this back-end. #1598
+- alpaka's memory fences can now also be applied to the grid level #1641
+- `alpaka::getWarpSize()` was renamed to `alpaka::getWarpSizes()` and will now return a `std::vector` of supported warp sizes #1644
+- Documentation updates:
+  - Improved installation and usage documentation #1571
+  - Added documentation on how to write unit tests #1609
+  - The HIP portion of the compiler support matrix has been simplified #1637
+
+### Misc:
+- The CUDA and HIP back-ends no longer explicitly set the device where this is unnecessary #1515
+- `clang-tidy`'s modernization suggestions have been applied to the code base #1584
+- alpaka's math headers have been squashed. For each back-end there is now only one header instead of one for each math function. #1585
+- Updated the Boost predefinition header to reflect the upgrade to Boost 1.74 #1586
+- Removed the `alpaka::extent` namespace (the contents now live in the main `alpaka` namespace) #1593
+- Refactored implementations of `BufCpu` and `BufUniformCudaHipRt` #1608
+- Removed unnecessary specializations of `GetPitchBytes` trait #1614
+- All alpaka-specific CMake variables follow the `${PROJECT_NAME}_VARIABLE_FOO_BAR` pattern. This means that all alpaka-specific CMake variables look like this: `alpaka_VARIABLE_FOO_BAR`. #1653
+- alpaka now enforces that kernel arguments are trivially copyable #1635
+- Renamed namespace `traits` to `trait` #1651
```

Comment:
```suggestion
```
Since you mention it a second time later.

Reply:
Yes, this is done on purpose so all breaking changes are grouped in one place.